Don't merge before release: 3.13 blog post

kennethlarsen opened the pull request:
Ready for final review

kennethlarsen committed:
Add template for 3.13 blog post

kennethlarsen committed:
add LTS info and prepare feature content

kennethlarsen committed:
WIP

kennethlarsen committed:
Text for all ember features

kennethlarsen committed:
Final edits

kennethlarsen committed:
Merge branch 'master' into release/3-13-post

kennethlarsen committed:
Typo

chancancode commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+Additionally, versions 3.12 of Ember and Ember Data are now promoted to LTS, which stands for Long Term Support. An LTS version of Ember continues to receive security updates for 9 release cycles (54 weeks) and bugfixes for 6 cycles (36 weeks). LTS releases typically occur every four minor versions. The previous LTS versions for Ember and Ember Data were 3.4.
+
+If you are upgrading from the previous LTS version to 3.12 then you are getting access to these features:
```
When I first read this, I thought it was saying 3.6, 3.8, 3.10, 3.11 are all LTS versions, which confused me.

chancancode replied:
(I think the list is good/interesting, but the way it is worded/presented was a bit confusing to me)

chancancode commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+If you're not an addon author or maintain a component manager then chances are that you will not need use this new feature.
+
+**Component-class generator (4 of 5)**
```
```suggestion
**`component-class` generator (4 of 5)**
```

chancancode commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+**Component-class generator (4 of 5)**
+
+When generating a component using `ember generate component modal` it will now by default use the `--no-component-class` flag. This means that when you generate a component it will only generate the template file and the test file.
```
This is only true in Octane apps

chancancode commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+When generating a component using `ember generate component modal` it will now by default use the `--no-component-class` flag. This means that when you generate a component it will only generate the template file and the test file.
+
+If you want a component class file when generating the component you can use the `--with-component-class` flag.
```
```suggestion
In the cases where you need a component class file, you can either pass the `--with-component-class` flag to the `component` generator, or add one after the fact with the new `ember generate component-class modal` generator.
```

chancancode commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+With component templates co-location it is now possible to place a component's JavaScript file and it's template in the same directory on the file system.
+
+[The RFC](https://github.com/emberjs/rfcs/blob/master/text/0481-component-templates-co-location.md) contains a detailed explanaition for this feature.
```
Only the low-level APIs: `setComponentTemplate()`, `getComponentTemplate()` and `templateOnlyComponent()` are enabled in this release. The supporting code for the actual co-location feature is included in the release, but is only enabled with octane apps... this feature will be enabled for all apps in the next version.

chriskrycho replied:
@chancancode @kennethlarsen see my recommendation in the overall review comment I'm leaving; I'd suggest we note that users can try it out by testing Octane on the beta release channel.

chriskrycho commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+---
+title: Ember 3.13 Released
+author: Kenneth Larsen
+tags: Releases, 2019, 3, 3.13
```
Needs `Version 3.x` in this list so it shows up under the 3.x link on the blog!

```suggestion
tags: Releases, 2019, 3, 3.13, Version 3.x
```

kennethlarsen replied:
omg thanks!

chriskrycho replied:
Added this as a suggestion so you can trivially auto-apply it!

chriskrycho commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+#### New Features (5)
+
+**Tracked Properties and Tracked Property Updates (1 of 5)**
```
Suggestion for semantics: these are being **strongly emphasized** but what you really mean is "This is a section." These should be `#####` for `h5` tags. We need to add styles for that to look right, which I can certainly do if someone points me to where (and I can go back and update all previous blog posts using this format as well).

kennethlarsen replied:
You're right. It's been there because we don't have the styles for it, I believe. There is a redesign going on right now which I hope will resolve the issue

chriskrycho commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,150 @@
+
+Tracked properties introduce a simpler and more ergonomic system for tracking state change in Ember applications. By taking advantage of new JavaScript features, tracked properties allow Ember to reduce its API surface area while producing code that is both more intuitive and less error-prone.
+
+During the Ember Octane preview period we encountered some issues with the current design for Tracked Properties that was proposed and accepted in [RFC 410](https://github.com/emberjs/rfcs/blob/master/text/0410-tracked-properties.md). The primary issues were specifically around interop between tracked properties, computed properties, and autotracking, with a few extra issues and inconsistencies s ...
```
I believe there were also, and equally seriously, issues around interop with sync observers—and the fixes made with them have performance implications that people should know about here. cc. @pzuraq.

pzuraq replied:
Yes, observer heavy applications should definitely keep an eye out for performance regressions with this release, and consider switching on the async-observers optional feature in order to mitigate if they find issues. We should note that here for sure.

chancancode replied:
Should we just add async observers as a separate top-level items?

chriskrycho reviewed (changes requested):
Had a couple small suggestions here, but I have one *big* suggestion: this is the last release before we flip the switch on Octane, and I think it'd be great if we said as loudly as possible ***THE NEXT RELEASE IS OCTANE, SO EVEN IF YOU DON'T NORMALLY TEST BETAS PLEAAAAASE TEST THE 3.14 BETA*** with a link to the Octane guides for how to test it out. 😄

chancancode commented:
@chriskrycho I think the plan is people should be able to test out Octane on this stable release by enabling the flag, since all the features are in this release already.

tomdale commented:
@kennethlarsen Would you be open to @wycats and I adding an introductory section to this blog post setting the context for 3.13 and Octane?

kennethlarsen commented:
@tomdale I think that would be great!

kennethlarsen committed:
Implement feedback

tomdale committed:
Adds Octane-specific context to the 3.13 release blog post

As the release before Octane, 3.13 contains significant new features that are part of the Octane programming model. This commit updates the post to expand on the relationship of 3.13 to Octane, describing it as the Octane preview, and provides guidance on how people should be thinking about adopting new features.

tomdale commented:
@kennethlarsen Awesome! @wycats and I put up our changes in a separate PR for you to review: https://github.com/ember-learn/ember-blog/pull/282.

tomdale committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Godfrey Chan <godfreykfc@gmail.com>

tomdale committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Godfrey Chan <godfreykfc@gmail.com>

tomdale committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Godfrey Chan <godfreykfc@gmail.com>

kennethlarsen committed:
Octane-related 3.13 blog post revisions (#282)

Octane-related 3.13 blog post revisions

kennethlarsen committed:
Make markdownlint happy

dmzza commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,188 @@
+
+This release kicks off the 3.14 beta cycle for all sub-projects. We encourage our community (especially addon authors) to help test these beta builds and report any bugs before they are published as a final release in six weeks' time. The [ember-try](https://github.com/ember-cli/ember-try) addon is a great way to continuously test your projects against the latest Ember releases.
+
+Among other things, the [`@model` named argument](https://github.com/emberjs/rfcs/blob/master/text/0523-model-argument-for-route-templates.md) will be available to route templates. While not officially an "Octane feature", this feature will available for the first time in the 3.14 release, which coincides with the Octane release. Since the Octane guides and documentation are targeting the same ve ...
```
```suggestion
Among other things, the [`@model` named argument](https://github.com/emberjs/rfcs/blob/master/text/0523-model-argument-for-route-templates.md) will be available to route templates. While not officially an "Octane feature", this feature will available for the first time in the 3.14 release, which coincides with the Octane release. Since the Octane guides and documentation are targeting the same version, you will see them using this feature, but please note that the feature is not available in the 3.13 Octane preview.
```

Gaurav0 commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,188 @@
+// .ember-cli.js
+const { setEdition } = require('@ember/edition-utils');
+
+setEdition('octane');
```
Precisely what does this do?

jenweber replied:
I think this switches the blueprints. If that's true, we should specifically say that.

kennethlarsen committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: David Mazza <davidedwardmazza@gmail.com>

kennethlarsen committed:
Add proper tag

kennethlarsen committed:
Merge branch 'release/3-13-post' of github.com:ember-learn/ember-blog into release/3-13-post

wycats committed:
Update 2019-09-20-ember-3-13-released.md

wycats committed:
Update 2019-09-20-ember-3-13-released.md

jenweber commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+Today, the Ember project is releasing version 3.13 of Ember.js, Ember Data, and Ember CLI.
+
+Ember 3.13 is the final release of Ember before Octane. Ember 3.13 is also the Octane preview. The preview allows adventurous community members to help us put the final round of polish on the edition before Octane becomes official in 3.14.
```
```suggestion
Ember 3.13 is the final release of Ember before Octane. Ember 3.13 is also the Octane preview. The preview allows adventurous community members to start using its stable features and help put the final round of polish on supporting tools and documentation before Octane becomes official in 3.14. You can learn more about what to expect from Octane and how to prepare your apps in a [previous blog post](https://blog.emberjs.com/2019/08/15/octane-release-plan.html).
```

jenweber commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+While these APIs are stable, not all Octane features are compatible with popular addons yet. We are also still updating documentation, and important tools like the Ember Inspector do not yet work with some Octane features.
+
+In Ember 3.14, Octane will be the recommended way to use Ember and new apps will have Octane's optional features enabled by default. The [guides](https://octane-guides-preview.emberjs.com/release/) and [tutorial](https://octane-guides-preview.emberjs.com/release/tutorial/00-part-1/) will show Octane examples, and codemods will be available to help users migrate to Octane.
```
```suggestion
Then, when Ember 3.14 arrives, Octane will be the recommended way to use Ember and new apps will have Octane's optional features enabled by default. The guides and tutorials will show Octane examples, and codemods will be available to help users migrate to Octane. Feel free to take a look at the [current draft](https://emberjs.com/editions/octane) of these learning resources, which are deployed to a temporary URL.
```

jenweber commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+Tracked properties introduce a simpler and more ergonomic system for tracking state change in Ember applications. Tracked properties allow Ember to reduce its API surface area while producing code that is both more intuitive and less error-prone.
+
+Have a look at the [guides] to learn more about tracked properties. For design details, check out [RFC 410](https://github.com/emberjs/rfcs/blob/master/text/0410-tracked-properties.md) and [RFC 478](https://github.com/emberjs/rfcs/blob/master/text/0478-tracked-properties-updates.md).
```
```suggestion
Have a look at the [guides] to learn more about tracked properties. For design details, check out [RFC 410](https://github.com/emberjs/rfcs/blob/master/text/0410-tracked-properties.md) and [RFC 478](https://github.com/emberjs/rfcs/blob/master/text/0478-tracked-properties-updates.md).
```
This should link to the API docs for tracked, ideally. If we have to, we can link to https://emberjs.com/editions/octane and say "have a look at the draft guides...."

jenweber commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+## Ember 3.12 is an LTS Release
+
+If you are upgrading from the previous LTS version (3.8) to 3.12 then you are getting access to these features:
```
love this section, thank you!

jenweber commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+So why is this a preview and not the final Octane release?
+
+While these APIs are stable, not all Octane features are compatible with popular addons yet. We are also still updating documentation, and important tools like the Ember Inspector do not yet work with some Octane features.
```
```suggestion
A big part of Octane is about delivering an excellent developer experience, whether you are a beginner or a power user. While these APIs are stable, not all Octane features are compatible with popular addons yet. We are also still updating documentation, and important tools like the Ember Inspector do not yet work with some Octane features. Octane is not complete until these supporting pieces are in place.
```

jenweber commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+In Ember 3.14, Octane will be the recommended way to use Ember and new apps will have Octane's optional features enabled by default. The [guides](https://octane-guides-preview.emberjs.com/release/) and [tutorial](https://octane-guides-preview.emberjs.com/release/tutorial/00-part-1/) will show Octane examples, and codemods will be available to help users migrate to Octane.
+
+The Ember tutorial has already been completely rewritten for Octane, and the easiest way to understand the big picture of Octane, especially if you're already an experienced Ember developer, is to [work through the new tutorial](https://octane-guides-preview.emberjs.com/release/tutorial/00-part-1/).
```
```suggestion
The Ember tutorial has already been completely rewritten for Octane, and the easiest way to understand the big picture of Octane, especially if you're already an experienced Ember developer, is to [work through the new tutorial](https://emberjs.com/editions/octane).
```

We should not link to the preview URL, since it will be removed when Octane ships.

kennethlarsen committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Jen Weber <j@jenweber.me>

kennethlarsen committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Jen Weber <j@jenweber.me>

kennethlarsen committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Jen Weber <j@jenweber.me>

kennethlarsen committed:
Update source/2019-09-20-ember-3-13-released.md

Co-Authored-By: Jen Weber <j@jenweber.me>

Gaurav0 commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+
+If you are upgrading from the previous LTS version (3.8) to 3.12 then you are getting access to these features:
+
+* `router` service 
```
The router service has been included since Ember 2.12

Suggest saying specifically how it was improved.

Gaurav0 commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+If you are upgrading from the previous LTS version (3.8) to 3.12 then you are getting access to these features:
+
+* `router` service 
+* Support for native classes
```
ES classes have been supported since August 2017, when [RFC 240](https://github.com/emberjs/rfcs/pull/240) was accepted.

Suggest saying [Native Class Constructor Update RFC](https://github.com/emberjs/rfcs/blob/master/text/0337-native-class-constructor-update.md) implemented

pzuraq replied:
Native classes were not officially supported until v3.6 (RFC implementation was incomplete + redesign occurred, no blog post announcement, no API docs). 3.10 adds decorators, which made native classes viable/recommendable for most users, so maybe it should say "Support for native classes + decorators"

Gaurav0 commented on source/2019-09-20-ember-3-13-released.md:
```diff
@@ -0,0 +1,190 @@
+* Element Modifier Manager (and modifiers in general)
+* Nested angle bracket component invocation
+* Angle bracket invocation for built-in components
+* Support for native decorators
```
I don't think any browser has native decorators. Reword?

jenweber reviewed (approved):
I chatted with Kenneth, and he says this is good to merge. I gave it a quick read through in the preview app and it all looks goo!

jenweber committed:
Update release blog post date

jenweber committed:
Merge branch 'master' into release/3-13-post
